File: static/js/changelog.js
const list = document.querySelectorAll("div.version>span.version_header>span.extend");
for (const el of list) {
    el.onclick = ev => {
        let version_div = ev.target.parentNode.parentNode;

        const extend = version_div.firstChild.lastChild;
        const to_togg = version_div.lastChild;
        console.log(to_togg.classList);
        if (to_togg.classList.contains("hide")) {
            to_togg.classList.remove("hide");
            to_togg.classList.add("animate");
            extend.classList.add("hide");
            extend.innerText = "-";
        } else {
            to_togg.classList.remove("animate");
            to_togg.classList.add("animate");
            to_togg.classList.add("hide");
            extend.classList.remove("hide");
            extend.innerText = "+";

        }
    }
}

File: src/changelog.rs
use actix_web::{get, HttpResponse, Responder};
use maud::html;

use crate::{meta, svgs::Svgs, VERSIONS};

#[get("/changelog.robson")]
pub async fn changelog() -> impl Responder {
  let html = html! {
    (meta())
    title { "Change Log" }
    script src="/static/js/changelog.js" defer {}
    link rel="stylesheet" href="/static/css/changelog.css";


    div id="root" {
      header {
        div.lg.animate onclick="window.location.href='/'" {
          (Svgs::Logo)
        }
        h1{ "Change Log"}
      }
      main {
        div.list {
          @for version in VERSIONS {
            div.version {
              span.version_header {
                h3 { (version.ver) }
                span.line {}
                span.extend { "+" }
              }
              div.log.hide {
                h4 { "Added" }
                div.features {
                  @for feature in version.features {
                    div.feature { (feature) }
                  }
                }
                h4 { "Changes" }
                div.changes {
                  @for change in version.changes {
                    div.change { (change) }
                  }
                }
              }
            }
          }
        }
      }
    }

  };
  HttpResponse::Ok().body(html.into_string())
}
